src/utils/index.ts

```diff
@@ -0,0 +1,59 @@
+import type { Data, Inputs, AttributeVal, HtmlAttributes } from '../types';
+
+// Add all required search params with user inputs as values
+function formatUrl(url: string, params: string[], args: Inputs) {
+  if (!params || !args) return url;
+
+  let newUrl = new URL(url);
+
+  params.forEach((param: string) => {
+    if (args[param]) newUrl.searchParams.set(param, args[param]);
+  });
+
+  return newUrl.toString();
+}
+
+// Construct HTML element and include all default attributes and user-inputted attributes
+function createHTML(element: string, attributes: HtmlAttributes, args: Inputs) {
```

Comment:
Let's add a unit test that we're generating the right HTML?

Reply:
Added a unit tests file with a few tests for this function